nomad/structs/secure_variables.go

```diff
@@ -28,6 +28,65 @@ type SecureVariableData struct {
 	KeyID string // ID of root key used to encrypt this entry
 }
 
+func (sv SecureVariableData) Copy() *SecureVariableData {
+	out := make([]byte, 0, len(sv.Data))
+	copy(out, sv.Data)
+	return &SecureVariableData{
+		Data:  out,
+		KeyID: sv.KeyID,
+	}
+}
+
+func (sv SecureVariable) Copy() SecureVariable {
+	out := SecureVariable{
+		Namespace:   sv.Namespace,
+		Path:        sv.Path,
+		CreateIndex: sv.CreateIndex,
+		CreateTime:  sv.CreateTime,
+		ModifyIndex: sv.ModifyIndex,
+		ModifyTime:  sv.ModifyTime,
+	}
+	if sv.UnencryptedData != nil {
+		out.UnencryptedData = make(map[string]string, len(sv.UnencryptedData))
+		for k, v := range sv.UnencryptedData {
+			out.UnencryptedData[k] = v
+		}
+	}
+	if sv.EncryptedData != nil {
+		out.EncryptedData = sv.EncryptedData.Copy()
+	}
+	return out
+}
+
+func (sv SecureVariable) AsStub() SecureVariableStub {
```

Comment:
Super nitpicky for consistency: this method is always called `Stub()` in the rest of the code base.